Brokerages/Binance/BinanceSymbolMapper.cs

```diff
@@ -0,0 +1,268 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Brokerages.Binance
+{
+    /// <summary>
+    /// Provides the mapping between Lean symbols and Binance symbols.
+    /// </summary>
+    public class BinanceSymbolMapper : ISymbolMapper
+    {
+        /// <summary>
+        /// Symbols that are both active and delisted
+        /// </summary>
+        public static List<Symbol> KnownSymbols
+        {
+            get
+            {
+                var symbols = new List<Symbol>();
+                var mapper = new BinanceSymbolMapper();
+                foreach (var tp in KnownSymbolStrings)
+                {
+                    symbols.Add(mapper.GetLeanSymbol(tp, mapper.GetBrokerageSecurityType(tp), Market.Binance));
+                }
+                return symbols;
+            }
+        }
+
+        /// <summary>
+        /// The list of known Binance symbols.
+        /// https://api.Binance.com/v1/symbols
+        /// </summary>
+        public static readonly HashSet<string> KnownSymbolStrings = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
Is there a way to resolve this information on launch, otherwise we'll always be out of date with supported symbols. Also, in some cases, symbols might not be the same on different exchanges. For example, there's multiple tokens that trade under the ticker `HOT` -- this shouldn't cause issues when running a single brokerage, but can potentially cause SID collisions.

Reply:
Not really. We discussed it months ago and this feature isn't implemented yet. It would be nice to have a separate class to fetch all up-to-date static data on init.

Reply:
I think the real issue is that it is **not** static data. Crypto exchanges are notorious for adding/removing symbols quite frequently. So if we're running a live algorithm we'll need this data to (potentially) produce symbol delisted events and maybe (eventually) an IPO-like event (symbol listed?) I'm happy to keep this as static data as a first revision solution, but we'll need to come up with a good pattern for managing the additions/removals from crypto exchanges.